tests/NTest/NTestEnv.lua

```diff
@@ -0,0 +1,60 @@
+-- A series of convenient utility functions for use with NTest on NodeMCU
+-- devices under test (DUTs).
+
+local NTE = {}
+
+function NTE.hasC(m)
+  -- this isn't great, but it's what we've got
+  local mstr = node.info('build_config').modules
+  return mstr:match("^"..m..",") -- first module in the list
+      or mstr:match(","..m.."$") -- last module in the list
+      or mstr:match(","..m..",") -- somewhere else
+      or error("Missing C module " .. m)
+end
+
+function NTE.hasL(m)
+  return pcall(require, m) or error("Missing Lua module " .. m)
+end
+
+function NTE.hasEnv(reqFeats)
+  -- Stream through our configuration file and extract the features attested
+  --
+  -- We expect the configuration file to attest features as keys in a dictionary
+  -- so that they can be efficiently probed here but also so that we can
+  -- parameterize features.
+  --
+  -- { "features" : {
+  --    "feat1" : ... ,
+  --    "feat2" : ...
+  --   }
+  -- }
+  local decoder = sjson.decoder({
+    metatable = {
+      checkpath = function(t, p)
+        if #p > 2 then return true end              -- something we're keeping
+        if #p == 0 then return true end             -- root table
+        if p[1] == "features" then
+          if #p == 1 then return true end           -- features table
+          local thisFeat = p[2]
+          assert type(thisFeat) == "string"
+          for k, v in ipairs(reqFeats) do
+            if v == thisFeat then return true end   -- requested feature
+          end
+        end
+        return false
+      end
+    }
+  })
+  local cfgf = file.open("testenv.conf", "r")
+  local cstr
+  repeat cstr = cfgf:read(); decoder:write(cstr) until #cstr == 0
+  cfgf:close()
+  local givenFeats = decoder:result().features
+
+  local res = {}
+  for k, v in ipairs(reqFeats) do
+    res[v] = givenFeats[v] or error("Missing required feature " .. v)
```

Comment:
no need to copy into another array here. If all is there just return reqFeats

Reply:
But `reqFeats` doesn't have the associated configuration values?  Sorry, I must be missing something.

Reply:
Sorry, I meant return `givenFeats` which should be filtered already.